torchx/lib/torchx/backend.ex

```diff
@@ -291,14 +328,21 @@ defmodule Torchx.Backend do
     do: Nx.backend_transfer(tensor, TB) |> to_ref()
 
   # Update out tensor type here to mark the cases where our type policy mismatches with the libtorch's one.
-  defp to_tensor(ref, %T{type: type} = t) do
+  defp to_tensor(ref, %T{type: type, shape: shape} = t) do
```

Comment:
What do you think about adding another flag such as `Application.get_env(:torchx, :check_backend_shape_and_type, false)` that we set to true only for tests? So we only run those checks during development? :)